src/chains/ethereum/block/src/block.ts

```diff
@@ -41,13 +51,42 @@ export class Block {
       // TODO: support actual uncle data (needed for forking!)
       // Issue: https://github.com/trufflesuite/ganache/issues/786
       // const uncles = deserialized[2];
-      const totalDifficulty = deserialized[3];
+
+      let totalDifficulty: Buffer;
+      // if there are 7 serialized fields we are after shanghai
+      // as in shanghai we added `withdrawals` to the block data
+      if (deserialized.length === 7) {
+        this._rawWithdrawals = deserialized[3] || []; // added in Shanghai
+        totalDifficulty = deserialized[4];
+        this._rawTransactionMetaData = deserialized[5] || [];
+        this._size = Quantity.toNumber(deserialized[6]);
+      } else {
+        this._rawWithdrawals = null;
+        totalDifficulty = deserialized[3] as any;
+        this._rawTransactionMetaData = (deserialized[4] || []) as any;
+        this._size = Quantity.toNumber(deserialized[5] as any);
+      }
       this.header = makeHeader(this._raw, totalDifficulty);
-      this._rawTransactionMetaData = deserialized[4] || [];
-      this._size = Quantity.toNumber(deserialized[5]);
     }
   }
 
+  /**
+   * Migrates a serialized Block to the latest version. This should only be
+   * called on serialized data from blocks created before v7.8.0.
+   *
+   * This migration updates the `size` value of the block to the correct value
+   * by re-serializing the block for storage in the db.
+   * @param serialized
+   * @returns
+   */
+  static migrate(serialized: Buffer) {
+    const deserialized = decode<GanacheRawBlock>(serialized);
+    return serialize(
+      deserialized.slice(0, 3) as Head<EthereumRawBlock>,
```

Comment:
I don't think this is super critical, but as far as I can tell the `Head<T>` here is to remove the `withdrawals` from `EthereumRawBlock`, but unless I'm mistaken, it's already optional, based on the definition of `EthereumRawBlock`?

https://github.com/trufflesuite/ganache/pull/4272/files#diff-97cedef4b674bd7c1fceaad9d0b9dde6d59bd8ce65756c0bac54dedab35900bdR66
```
export type EthereumRawBlock = _EthereumRawBlock | Head<_EthereumRawBlock>;
```